integration/otlp_ingestion_test.go

```diff
@@ -186,3 +187,166 @@ func testOTLPIngestion(t *testing.T, enableSuffixes bool) {
+					]
+				},
+				"scopeMetrics": [
+					{
+						"metrics": [
+							{
+								"name": "explicit_bucket_histogram_series",
+								"description": "Explicit bucket histogram series",
+								"unit": "s",
+								"histogram": {
+									"aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
+									"dataPoints": [
+										{
+											"attributes": [
+												{ "key": "metric-attr", "value": { "stringValue": "metric value" } }
+											],
+											"bucketCounts": [0, 4, 3, 0, 3],
+											"explicitBounds": [0, 5, 10, 15],
+											"count": 10,
+											"sum": 20,
+											"timeUnixNano": "%d"
+										}
+									]
+								}
+							},
+							{
+								"name": "exponential_histogram_series",
+								"description": "Exponential histogram series",
+								"unit": "s",
+								"exponentialHistogram": {
+									"aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
+									"dataPoints": [
+										{
+											"attributes": [
+												{ "key": "metric-attr", "value": { "stringValue": "metric value" } }
+											],
+											"scale": 0,
+											"count": 15,
+											"sum": 25,
+											"zeroCount": 1,
+											"positive": {
+												"offset": 1,
+												"bucketCounts": [1, 0, 3, 2, 0, 3]
+											},
+											"negative": {
+												"offset": 0,
+												"bucketCounts": [4, 0, 0, 1]
+											},
+											"timeUnixNano": "%d"
+										}
+									]
+								}
+							}
+						]
+					}
+				]
+			}
+		]
+	}`, nowUnix, nowUnix))
+
+	res, err := c.PushOTLPPayload(jsonPayload, "application/json")
+	require.NoError(t, err)
+	require.Equal(t, 200, res.StatusCode)
+
+	// Check metric to track OTLP requests
+	require.NoError(t, mimir.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"cortex_distributor_otlp_requests_total"}, e2e.WithLabelMatchers(
+		labels.MustNewMatcher(labels.MatchEqual, "user", "user-1"))))
+
+	// Query the histogram series
+	if enableExplicitHistogramToNHCB {
+		// Verify that when flag is disabled, OTel explicit bucket histograms are converted to native histograms with custom buckets
+		result, err := c.Query("explicit_bucket_histogram_series", now)
+		require.NoError(t, err)
+		require.Equal(t, result.(model.Vector)[0].Histogram.Count, model.FloatString(10))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Sum, model.FloatString(20))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[0].Upper, model.FloatString(5))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[0].Count, model.FloatString(4))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[1].Upper, model.FloatString(10))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[1].Count, model.FloatString(3))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[2].Upper, model.FloatString(math.Inf(1)))
+		require.Equal(t, result.(model.Vector)[0].Histogram.Buckets[2].Count, model.FloatString(3))
```

Comment:
nit, this would be a little bit nicer:
```suggestion
		require.Equal(t, result.(model.Vector)[0].Histogram, &model.SampleHistogram{
			Count: model.FloatString(10),
			Sum:   model.FloatString(20),
			Buckets: model.HistogramBuckets{
				{
					Lower: model.FloatString(0),
					Upper: model.FloatString(5),
					Count: model.FloatString(4),
				},
				{
					Lower: model.FloatString(5),
					Upper: model.FloatString(10),
					Count: model.FloatString(3),
				},
				{
					Lower: model.FloatString(15),
					Upper: model.FloatString(math.Inf(1)),
					Count: model.FloatString(3),
				},
			},
		})
```